app/models/user.rb

```diff
@@ -110,4 +125,13 @@ def default_email_record
     result ||= valid_emails.first
     result
   end
+
+  # Custom time_zone validation as +#time_zone+ getter has been modified.
+  #
+  # Possible time_zones include +nil+ or those listed in ActiveSupport::TimeZone
+  def validate_time_zone
+    return if self[:time_zone].nil? || ActiveSupport::TimeZone.all.map(&:name).include?(self[:time_zone])
```

Comment:
For this and `#time_zone`, is it better to do `ActiveSupport::TimeZone[self[:time_zone]].present?` ?

Reply:
Definitely cleaner. Fixed and thanks!